Python/optimizer_analysis.c

```diff
@@ -588,16 +588,16 @@ remove_globals(_PyInterpreterFrame *frame, _PyUOpInstruction *buffer,
     INST->oparg = ARG;            \
     INST->operand = OPERAND;
 
+#define OUT_OF_SPACE_IF_NULL(RES, EXPR) \
```

Comment:
Expanding macro args more than once can be problematic, could you use a single argument?
```C
#define OUT_OF_SPACE_IF_NULL(EXPR) \
do { \
    if ((EXPR) == NULL) { \
        goto out_of_space;              \
    } \
while (0)
```
Used as follows:
```C
OUT_OF_SPACE_IF_NULL(attr = sym_new_known_notnull(ctx));
```